components/x-topic-search/src/lib/get-suggestions.js

```diff
@@ -3,39 +3,47 @@ const addQueryParamToUrl = (name, value, url, append = true) => {
 	return append === true ? `${url}&${queryParam}` : `${url}?${queryParam}`;
 };
 
-const suggest = function (suggestions, followedTopics, searchTerm) {
+const separateFollowedAndNot = (suggestions, followedTopics) => {
+	const matchingFollowedTopics = [];
+	const matchingUnfollowedTopics = [];
```

Comment:
I think some of the conditions can be removed by taking a slightly different approach to this:

```javascript
const separateFollowedAndUnfollowed = (suggestions, followedTopics) => {
	const followedSuggestions = suggestions.filter(suggestion => followedTopics.some(topic => topic.uuid === suggestion.id));
	const unfollowedSuggestions = suggestions.filter(suggestion => !followedTopics.some(topic => topic.uuid === suggestion.id));

	return { followedSuggestions, unfollowedSuggestions };
};

const suggest = function (suggestions, followedTopics) {
	suggestions.forEach((suggestion) => {
		if (suggestion && !suggestion.url){
			// TODO App needs different url?
			suggestion.url = '/stream/' + suggestion.id;
		}
	});

	return separateFollowedAndUnfollowed(suggestions, followedTopics);
};
```